cmd/agent/main.go

```diff
@@ -1,3 +1,303 @@
 package main
 
-func main() {}
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"math/rand"
+	"models"
+	"net/http"
+	"os"
+	"os/signal"
+	"regexp"
+	"runtime"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var url string = "http://"
+var pollInterval int = 2
+var reportInterval int = 10
+
+const validateAdderPattern string = "^[a-zA-Z/ ]{1,100}:[0-9]{1,10}$"
+
+const contentTypeSendMetric string = "text/plain"
+const minRandomValue float64 = 1.0
+const maxRandomValue float64 = 999.0
+
+type Options struct {
+	PORT           string
+	reportInterval int
+	pollInterval   int
+}
+
+/*type responseData struct {
+	r   *http.Response
+	err error
+}*/
+
+var wg sync.WaitGroup
+var m models.Monitor
+var options Options
+var httpClient *http.Client
+
+// var dataChannel chan responseData
+var gauges []models.Gauge
+var counters map[string]models.Counter
+
+func main() {
+
+	err := initialization()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	wg.Add(1)
+	go collectMetrics()
+	wg.Add(1)
+	go sendMetrics()
+	wg.Wait()
+}
+
+func collectMetrics() {
+
+	defer wg.Done()
+
+	channelCancel := make(chan os.Signal, 1)
+	signal.Notify(channelCancel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+
+	for {
+		select {
+		case <-channelCancel:
+			return
+		case <-time.After(time.Duration(pollInterval) * time.Second):
+			setValuesMonitor()
+		}
+	}
+}
+
+func sendMetrics() {
+
+	defer wg.Done()
+	channelCancel := make(chan os.Signal, 1)
+	signal.Notify(channelCancel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+
+	for {
+		select {
+		case <-channelCancel:
+			return
+		case <-time.After(time.Duration(reportInterval) * time.Second):
+			go doReqSendMetrics()
+			//case answer := <-dataChannel:
+			//	parseAnswer(&answer)
+		}
+	}
+}
+
+/*func parseAnswer(answer *responseData) {
+
+	err := answer.err
+	resp := answer.r
+	if err != nil {
+		fmt.Println("Error select:", err)
+	}
+	defer resp.Body.Close()
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("Error read data:", err)
+	}
+
+	fmt.Printf("Server Response: %s\n", responseBody)
+}*/
+
+func doReqSendMetrics() {
+
+	tmp_url := url + "/update/" + "counter/"
```

Comment:
[CHANGE REQUEST] Split the application into layers
https://github.com/golang-standards/project-layout
https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6#:~:text=The%20presentation%20layer%20is%20the,Update%20or%20Delete)%20operation%20flow.